src/main/java/com/adobe/cq/commerce/graphql/client/impl/GraphqlClientConfiguration.java

```diff
@@ -61,4 +65,22 @@
         description = "The maximum number of concurrent HTTP connections the connector can make",
         type = AttributeType.INTEGER)
     int maxHttpConnections() default MAX_HTTP_CONNECTIONS_DEFAULT;
+
+    @AttributeDefinition(
+        name = "Http connection timeout",
+        description = "The timeout in milliseconds for the http request",
```

Comment:
I'd reuse the text from the http client documentation: `The timeout in milliseconds until a connection is established. A timeout value of zero is interpreted as an infinite timeout.`

Reply:
Done